app/common/update.py

```diff
@@ -253,3 +258,65 @@ def merge_local_db():
 
         logger.info(str(records_inserted) + " records were inserted into local db.")
         logger.info(str(records_updated) + " records in local db were updated.")
+        
+
+def download_dat_files():
+    config = load_user_config()
+    valid_directory = False
+    
+    if not config["config"]["installdirectory"]:
+    
+        #First, let's check the default location and see if it's installed there
+        path = 'C:/Program Files (x86)/SquareEnix/DRAGON QUEST X/Game/Content/Data'
+        
+        if os.path.exists(path):
+            update_user_config('config', 'installdirectory', path)
+        else:
+            #Doesn't exist. Let's prompt the user for it.
+            warning_message(
+            title="[dqxclarity] Couldn't Find Directory",
+            message="Could not find DQX directory. Please select\n\nthe Data folder located in your\n\nDQX install location."
+        )
+            
+            path = askdirectory()
+            
+            #Path looks legit.
+            if 'Game/Content/Data' in path:
+                update_user_config('config', 'installdirectory', path)
+                valid_directory = True
+            else:
+                message_box_fatal_error(
+                    title="[dqxclarity] Invalid Directory",
+                    message="The path you provided is not a valid DQX path.\n\nPlease try again."
+                )
+    else:
+        if 'Game/Content/Data' in config["config"]["installdirectory"]:
+            valid_directory = True
+            
+    if valid_directory:
+        
+        #First, let's check if an idx file backup already exists
+        data_folder_path = config["config"]["installdirectory"]
+        idx_path = data_folder_path + "/data00000000.win32.idx.bak"
+        
+        #If it doesn't, let's back it up first
+        if not os.path.isfile(idx_path):
+            shutil.copy(data_folder_path + "/data00000000.win32.idx", data_folder_path + "/data00000000.win32.idx.bak")
+        
+        #Now let's download the files
+        try:
+            logger.info("Downloading DAT1 and IDX files.")
+            request = requests.get(GITHUB_CLARITY_DAT1_URL, timeout=15)
+            if request.status_code == 200:
+                with open(data_folder_path + "\data00000000.win32.dat1", "w+b") as f:
+                    f.write(request.content)
+            
+            request = requests.get(GITHUB_CLARITY_IDX_URL, timeout=15)
+            if request.status_code == 200:
+                 with open(data_folder_path + "\data00000000.win32.idx", "w+b") as f:
+                    f.write(request.content)
+        except Exception as e:
+            logger.error(f"Failed to download data files. Error: {e}")
+            input("Press ENTER to exit.")
+            sys.exit()        
+        

```

Comment:
We should use some type of `logger.success` messaging to let the user know the operation was successful.